### PR TITLE
Add container connection retry logic

### PR DESCRIPTION
**Pull request by spaceisfun**
Bug 29694123: Failed FactoryOrchestratorTest.FactoryOrchestratorTest.ExecuteTests: Test Run Failed.

Tasks were failing because the container wasn't ready at the start of test execution

### CODE REVIEW

**Review by patelp3989 — APPROVED** [review]

File: src/Service/ServiceExe.cs [hunk 1]
```diff
@@ -1791,42 +1791,58 @@ public async Task<bool> TryVerifyContainerConnection()
...
+                    if (previousContainerStatus == IsContainerConnected)
+                    {
+                        // Connection lost
+                        LogServiceEvent(new ServiceEvent(ServiceEventType.ContainerDisconnected, previousContainerGuid, Resources.ContainerDisconnected));
+                    }
 
-                throw;
+                    if (w.ElapsedMilliseconds * 1000 > retrySeconds)
+                    {
+                        throw;
```
Comment (girishbalaji) [inline src/Service/ServiceExe.cs:1843]: would we want to throw a timeout exception here?

**Review by girishbalaji — APPROVED** [review]

File: src/Service/ServiceExe.cs (hunk 1 above)
Comment (spaceisfun) [reply src/Service/ServiceExe.cs:1843]: No, I only want container APIs to throw exceptions of type FactoryOrchestratorContainerException. I think the fact that it timed out is more of an implementation detail that a consumer doesn't need to know.

Comment (girishbalaji) [reply src/Service/ServiceExe.cs:1843]: Ah gotcha, makes sense

File: src/Service/ServiceExe.cs [hunk 2]
```diff
@@ -1816,7 +1817,7 @@ public async Task VerifyContainerConnection(int retrySeconds = 0)
 
                     await ConnectToContainer();
 
-                    if (previousContainerStatus == IsContainerConnected)
+                    if (IsContainerConnected && (previousContainerStatus == IsContainerConnected))
```
Comment (girishbalaji) [inline src/Service/ServiceExe.cs:1820]: if (IsContainerConnected && previousContainerStatus) should be equivalent

**Review by girishbalaji — APPROVED** [review]

File: src/Service/ServiceExe.cs (hunk 2 above)
Comment (spaceisfun) [reply src/Service/ServiceExe.cs:1820]: derp you are right

**Comment by spaceisfun** [comment]
Sorry for the churn but I'd appreciate a final review, I think this looks a lot more reliable now. @girishbalaji

**Review by girishbalaji — APPROVED** [review]

**Pull request merged**